platform/ethereum/blockbook/transaction.go

```diff
@@ -0,0 +1,139 @@
+	if token != "" {
+		normalizedToken = Address.EIP55Checksum(token)
+	}
+	for _, srcTx := range srcPage.Transactions {
+		tx := normalizeTxWithAddress(&srcTx, normalizedAddr, normalizedToken, coinIndex)
+		txs = append(txs, tx)
+	}
+	return txs
+}
+
+func normalizeTx(srcTx *Transaction, coinIndex uint) blockatlas.Tx {
+	status, errReason := srcTx.EthereumSpecific.GetStatus()
+	normalized := blockatlas.Tx{
+		ID:       srcTx.TxID,
+		Coin:     coinIndex,
+		From:     srcTx.FromAddress(),
+		To:       srcTx.ToAddress(),
+		Fee:      blockatlas.Amount(srcTx.Fees),
+		Date:     srcTx.BlockTime,
+		Block:    normalizeBlockHeight(srcTx.BlockHeight),
+		Status:   status,
+		Error:    errReason,
+		Sequence: srcTx.EthereumSpecific.Nonce,
+	}
+	FillMeta(&normalized, srcTx, coinIndex)
+	return normalized
+}
+
+func normalizeTxWithAddress(srcTx *Transaction, address, token string, coinIndex uint) blockatlas.Tx {
+	normalized := normalizeTx(srcTx, coinIndex)
+	normalized.Direction = GetDirection(address, normalized.From, normalized.To)
+	FillMetaWithAddress(&normalized, srcTx, address, token, coinIndex)
+	return normalized
+}
+
+func normalizeBlockHeight(height int64) uint64 {
+	if height < 0 {
+		return uint64(0)
+	}
+	return uint64(height)
+}
+
+func FillMeta(final *blockatlas.Tx, tx *Transaction, coinIndex uint) {
+	if len(tx.TokenTransfers) == 1 {
+		transfer := tx.TokenTransfers[0]
+		final.Meta = blockatlas.TokenTransfer{
+			Name:     transfer.Name,
+			Symbol:   transfer.Symbol,
+			TokenID:  transfer.Token,
+			Decimals: transfer.Decimals,
+			Value:    blockatlas.Amount(transfer.Value),
+			From:     transfer.From,
+			To:       transfer.To,
+		}
+		return
+	}
+	fillMeta(final, tx, coinIndex)
+}
+
+func FillMetaWithAddress(final *blockatlas.Tx, tx *Transaction, address, token string, coinIndex uint) {
+	if len(tx.TokenTransfers) > 0 {
+		for _, transfer := range tx.TokenTransfers {
+			if transfer.To == address || transfer.From == address {
+				// filter token if specified
+				if token != "" {
+					if token != transfer.Token {
+						continue
+					}
+				}
+				direction := GetDirection(address, transfer.From, transfer.To)
+				metadata := blockatlas.TokenTransfer{
+					Name:     transfer.Name,
+					Symbol:   transfer.Symbol,
+					TokenID:  transfer.Token,
+					Decimals: transfer.Decimals,
+					Value:    blockatlas.Amount(transfer.Value),
+				}
+				if direction == blockatlas.DirectionSelf {
+					metadata.From = address
+					metadata.To = address
+				} else if direction == blockatlas.DirectionOutgoing {
+					metadata.From = address
+					metadata.To = transfer.To
+				} else {
+					metadata.From = transfer.From
+					metadata.To = address
+				}
+				final.Direction = direction
+				final.Meta = metadata
+				return
+			}
+		}
+	}
+	fillMeta(final, tx, coinIndex)
+}
+
+func fillMeta(final *blockatlas.Tx, tx *Transaction, coinIndex uint) {
+	gasUsed := tx.EthereumSpecific.GasUsed
+	if gasUsed != nil && gasUsed.Int64() == 21000 {
```

Comment:
I think it's a bug - ETH transfers not always 21000 gas limit

Reply:
this checks `GasUsed` not (GasLimit), plain ether transfer always costs 21000 gas, we can improve this once we can check `len(data)`